cpu.c

```diff
@@ -2101,12 +2107,20 @@ static void __gb_step_cpu(struct gb_s *gb)
         __gb_draw_line(gb);
 #endif
     }
+#endif
 }
 
+#ifdef DEBUG
+uint8_t gb_run_frame(struct gb_s *gb)
```

Comment:
`gb_run_frame` should be a public function what we assume it is available.

Reply:
I'm not sure if my modification are right, is this what you meaning?
```cpp
#ifndef DEBUG
void gb_run_frame(struct gb_s *gb)
{
    gb->gb_frame = 0;

    while (!gb->gb_frame)
        __gb_step_cpu(gb);
}
#else
uint8_t gb_run_frame(struct gb_s *gb)
{
    return __gb_step_cpu(gb);
}
#endif
```

Reply:
No, you have to make sure that the debug build (i.e. `-DDEBUG`) behaves the same as release build.

Reply:
You can change the internal state of `gb_frame` for the execution.

Reply:
since `gbz80_step` need cycles of instruction, i wandor any other way to retrive cycles information from `gb_run_frame` without return is possible?

Reply:
```diff
diff --git a/cpu.c b/cpu.c
index eb67ddb..da7206b 100644
--- a/cpu.c
+++ b/cpu.c
@@ -186,7 +186,7 @@ static uint8_t __gb_execute_cb(struct gb_s *gb)
 }
 
 /* Internal function used to step the CPU */
-static void __gb_step_cpu(struct gb_s *gb)
+static uint8_t __gb_step_cpu(struct gb_s *gb)
 {
     uint8_t opcode, inst_cycles;
     static const uint8_t op_cycles[0x100] = {
@@ -1954,6 +1954,9 @@ static void __gb_step_cpu(struct gb_s *gb)
         (gb->gb_error)(gb, GB_INVALID_OPCODE, opcode);
     }
 
+#ifdef DEBUG
+    gb->gb_frame = 1;
+#else
     /* DIV register timing */
     gb->counter.div_count += inst_cycles;
 
@@ -2101,12 +2104,17 @@ static void __gb_step_cpu(struct gb_s *gb)
         __gb_draw_line(gb);
 #endif
     }
+#endif
+    return inst_cycles;
 }
 
-void gb_run_frame(struct gb_s *gb)
+uint8_t gb_run_frame(struct gb_s *gb)
 {
+    uint8_t cycles;
     gb->gb_frame = 0;
 
     while (!gb->gb_frame)
-        __gb_step_cpu(gb);
+        cycles = __gb_step_cpu(gb);
+    return cycles;
 }
```

Reply:
The initialization of `gb_frame` should not be conditionally built.